### PR TITLE
`Development`: Remove unnecessary quiz deletion requests in cypress e2e tests

### PR DESCRIPTION
**Pull request by jhainzlschmid**
### Motivation and Context

The quiz exercise tests sometimes fail because a [request to the Artemis Server to delete a quiz exercise fails](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-QE-606/test/case/450278576). This is most likely a timing issue since we had a similar issue when deleting courses. However the deletion of the quiz exercises is not relevant for the tests themselves at all. The created exercises will be deleted together with the course later on anyways.

### Description

In this PR I removed the quiz deletion calls in the `QuizExerciseManagement` and in the `QuizExerciseParticipation` specs (this should remove the SQL constraint errors, reducing the flakiness of the tests). I also made some general code quality improvements and improved the login command because there seemed to be some synchronization issues. I did not refactor the `QuizExerciseAssessment` spec because it is currently failing and is being fixed in another PR.

### Steps for Testing

[Tests are executed by Bamboo (no new failures have been added by this PR)](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-613)

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by akesfeden — DISMISSED** [review]
Code changes look good to me.

File: src/test/cypress/integration/exercises/quiz-exercise/QuizExerciseManagement.spec.ts [hunk 1]
```diff
@@ -10,68 +10,46 @@ const admin = artemis.users.getAdmin();
...
 
         it('Creates a Quiz with Multiple Choice', () => {
-            quizCreation.addMultipleChoiceQuestion('MC Quiz' + uid);
-            quizCreation.saveQuiz().then((quizResponse) => {
-                quizExercise = quizResponse.response?.body;
-                cy.visit('/course-management/' + course.id + '/quiz-exercises/' + quizExercise.id + '/preview');
-                cy.contains('MC Quiz' + uid).should('be.visible');
-            });
+            quizCreation.addMultipleChoiceQuestion(quizQuestionTitle);
+            saveAndVerifyQuizCreation(quizQuestionTitle);
```
Comment (coaxial1) [inline src/test/cypress/integration/exercises/quiz-exercise/QuizExerciseManagement.spec.ts:42]: Seems like you're always using ``quizQuestionTitle`` as parameter for all calls of ``saveAndVerifyQuiyCreation``. I'd sugget removing the parameter of the method, and instead reference the ``quizQuestionTitle`` directly in ``saveAndVerifyQuiyCreation``.

**Review by coaxial1 — CHANGES_REQUESTED** [review]
Looks good to me!
Just one thing I'd change, but I'm open for discussion if there was a reason behind that :)

File: src/test/cypress/integration/exercises/quiz-exercise/QuizExerciseManagement.spec.ts (hunk 1 above)
Comment (jhainzlschmid) [reply src/test/cypress/integration/exercises/quiz-exercise/QuizExerciseManagement.spec.ts:42]: You are right. When I originally refactored it I first thought that each quiz type has to have its own specific title, but as it turned out that that didn't matter at all. I agree that it is better to use the constant directly in the helper method. I updated it. I also executed the test locally against TS3 and it still works.

**Review by coaxial1 — APPROVED** [review]

**Review by akesfeden — APPROVED** [review]
Approved again after changes.

**Pull request merged**